cpp/src/arrow/type.cc

```diff
@@ -2090,17 +2090,39 @@ Status SchemaBuilder::AreCompatible(const std::vector<std::shared_ptr<Schema>>&
   return Merge(schemas, policy).status();
 }
 
+std::vector<std::shared_ptr<Field>> make_fields(
+  const std::initializer_list<std::pair<std::string, std::shared_ptr<DataType>>>& init_list) {
+    std::vector<std::shared_ptr<Field>> fields;
+    fields.reserve(init_list.size());
+    for (const auto& pair : init_list) {
+        fields.push_back(std::make_shared<Field>(pair.first, pair.second));
+    }
+    return fields;
+}
+
 std::shared_ptr<Schema> schema(std::vector<std::shared_ptr<Field>> fields,
                                std::shared_ptr<const KeyValueMetadata> metadata) {
   return std::make_shared<Schema>(std::move(fields), std::move(metadata));
 }
 
+std::shared_ptr<Schema> schema(const std::initializer_list<std::pair<std::string, std::shared_ptr<DataType>>>& fields,
+                               std::shared_ptr<const KeyValueMetadata> metadata) {
+  return std::make_shared<Schema>(std::move(make_fields(fields)), std::move(metadata));
```

Comment:
`make_fields()` return a prvalue, not need to be moved.

Reply:
indeed, thank you